server/lib/two-factor-authentication.js

```diff
@@ -22,6 +22,20 @@ export function verifyTwoFactorAuthenticatorCode(encryptedTwoFactorAuthToken, tw
   return verified;
 }
 
+/** Verifies a submitted recovery code against the hashed ones saved for them in the DB.
+ * If the recovery code is valid, it returns the remaining recovery codes to update the user with.
+ * If the recovery code is not valid, it returns null so we can throw an error.
+ */
+export function verifyTwoFactorAuthenticationRecoveryCode(recoveryCode, hashedRecoveryCodes) {
+  const verified = hashedRecoveryCodes.includes(crypto.hash(recoveryCode));
```

Comment:
This will crash if `user.twoFactorAuthRecoveryCodes` is not defined. We should handle this with a proper error message.

Reply:
Have fixed this by checking in `server/controllers/users.js` that the recovery code is a string (since due to PR changes we now also use `crypto.hash` in there too).